core/presolve.gms

```diff
@@ -89,11 +89,149 @@ loop(regi,
 );
 display p_efFossilFuelExtr;
 
+
+***--------------------------------------
+***    MAgPIE coupling: run MAgPIE
+***--------------------------------------
+
+*** Decide whether MAgPIE should be executed. Also triggers the update of MAgPIE data in multiple locations.
+if(magpieIter(iteration) AND cm_MAgPIE_Nash eq 1,
+    sm_updateMagpieData = 1; !! Run MAgPIE in this Nash iteration
+else
+    sm_updateMagpieData = 0; !! Don't run MAgPIE in this Nash iteration
+);
+
+*** Run MAgPIE
+if (sm_updateMagpieData eq 1,
+*** Track runtime: done in mag2rem.R
+*** Temporarily change numeric round format (nr) and number of decimals (nd) of the
+*** outpt of the put_utility such that the arguments passed to mag2rem.R have integer format
+  sm_tmp  = logfile.nr;
+  sm_tmp2 = logfile.nd;
+  logfile.nr = 1;
+  logfile.nd = 0;
+  sm_magpieIter = sm_magpieIter + 1;
+  put_utility  "exec.checkErrorLevel" / "Rscript mag2rem.R " sm_magpieIter " " ord(iteration);
+  logfile.nr = sm_tmp;
+  logfile.nd = sm_tmp2;
+*** Track runtime
+  putclose runtime gyear(jnow):0:0 "-" gmonth(jnow):0:0 "-" gday(jnow):0:0 " " ghour(jnow):0:0 ":" gminute(jnow):0:0 ":" gsecond(jnow):0:0 ",GAMS," iteration.val:0;
+*** In coupled runs overwrite pebiolc production from look-up table with actual MAgPIE values
+*** Read production of 2nd gen. purpose grown bioenergy from MAgPIE (given to MAgPIE from previous Remind run)
+  Execute_Loadpoint 'magpieData.gdx' pm_pebiolc_demandmag;
+);
+
+*** At the beginning of each Nash iteration reset pm_macBaseMagpie to start values (from lookup table or from MAgPIE),
+*** since it gets changed by calculations further down in each Nash iteration, regardless of whether MAgPIE runs or not.
+if (sm_magpieIter gt 0,
+*** MAgPIE has run once at least: the start values for pm_macBaseMagpie come from the last MAgPIE iteration
+  Execute_Loadpoint 'magpieData.gdx' f_macBaseMagpie_coupling;
+  pm_macBaseMagpie(ttot,regi,emiMacMagpie(enty))$(ttot.val ge 2005) = f_macBaseMagpie_coupling(ttot,regi,emiMacMagpie);
+*** Biomass emission factor is set to zero after MAgPIE has run at least one, since biomass emissions are included in the emissions imported above. 
+  p_efFossilFuelExtr(regi,"pebiolc","n2obio") = 0.0;
+*DK* In coupling mode LU emissions are abated in MAgPIE (moved here from core/datainput.gms)
+  pm_macSwitch(ttot,regi,enty)$emiMacMagpie(enty) = 0;
+else 
+*** MAgPIE has not run (and might never run): the start values for pm_macBaseMagpie come from input data file
+  pm_macBaseMagpie(ttot,regi,emiMacMagpie(enty))$(ttot.val ge 2005) = f_macBaseMagpie(ttot,regi,emiMacMagpie,"%cm_LU_emi_scen%","%cm_rcp_scen%");
+);
+
+display p_efFossilFuelExtr;
+
+*** DK: moved here from core/datainput.gms, because pm_macSwitch is changed above after first MAgPIE iteration
```

Comment:
I think, we do not need initial any more. We have git-managemant and only the initials will not be understandable if people are leaving